crates/services/p2p/src/service.rs

```diff
@@ -186,6 +194,14 @@ where
             }
             p2p_event = self.p2p_service.next_event() => {
                 match p2p_event {
+                    Some(FuelP2PEvent::PeerInfoUpdated { peer_id, block_height }) => {
+                        let block_height_data = BlockHeightHeartbeatData {
+                            peer_id: peer_id.into(),
+                            block_height,
+                        };
+
+                        let _ = self.shared.block_height_broadcast.send(block_height_data);
```

Comment:
This is an example of where a height and therefor peers can be missed. Just send height only.

Reply:
I didn't get you=) The code sends the height and only **one** peer id corresponding to the received heights.

Reply:
I think this is leaking implementation details of how fuel-sync currently works into the p2p design. From p2p perspective it just reports which data it sees from which peers, it's up to other services to decide if the peer info is useful or not.